dot/types/grandpa.go

```diff
@@ -111,19 +111,32 @@ func (v *GrandpaVoter) Decode(r io.Reader) error {
 		return err
 	}
 
-	v.Key = key
+	v.Key = *key
 	v.ID = id
 	return nil
 }
 
+// Encode will encode the GrandpaVoter
+func (v *GrandpaVoter) Encode() ([]byte, error) {
+	enc := []byte{}
+	b := v.Key.Encode()
+	enc = append(enc, b...)
```

Comment:
does the new SCALE package work with custom encodings? for example, if `scale.Marshal(*GrandpaVoter)` is called, will this function be called by SCALE?

Reply:
No it does not. I also do not believe the GrandpaVoter is scale encoded because of the ed25519.PublicKey (Don't think the length is the leading byte of the encoded pubkey)

Reply:
yeah, it's always a `[]byte` of length 32. unfortunately the ed25519 has it just as `[]byte` instead of `[32]byte` so we can't directly SCALE encode it. maybe as a next step for SCALE we can add custom type encodings

Reply:
Yeah I think that could be a good idea

Reply:
this method should just be 
```
return scale.Marshal(v)
```
if you change `ed25519.PublicKey` to `[32]byte`

Reply:
I think we could just delete it all together and just directly marshal/unmarshal. Would also allow us to get rid of the custom encode and decode functions for GrandpaVoters

Reply:
I will look into this now actually. Could be a nice change to include in this PR

Reply:
if this is a huge change maybe do it in another PR, since I think the crypto/ed25519 golang lib we're using has some nice functions that we currently use (but maybe it's not a huge change)

Reply:
Yeah good call, I will do this in a separate PR